astropy/config/configuration.py

```diff
@@ -425,13 +432,13 @@ def reload(self):
             baseobj[self.name] = newobj[self.name]
         return baseobj.get(self.name)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return (
             f"<{self.__class__.__name__}: name={self.name!r} value={self()!r} at"
             f" 0x{id(self):x}>"
         )
 
-    def __str__(self):
+    def __str__(self) -> str:
```

Comment:
is this needed ? it seems that mypy is happy with leaving this annotation implicit: there's the same method, left un-annotated, in `Conf`. I don't mind the extra annotation, but I guess it should at least be applied (or left out) consistently.

Reply:
> it seems that mypy is happy with leaving this annotation implicit: 

Mypy does not check unannotated functions unless it is configured to do so, so if the return type annotation is removed then Mypy will be happy with this function regardless of what its contents are.

> there's the same method, left un-annotated, in `Conf`. 

This is how `Conf.__str__()` begins:
https://github.com/astropy/astropy/blob/2f43420d6ec891afdc5a9e2009938f290b4f9c0a/astropy/config/configuration.py#L91-L94
Mypy doesn't like that because the type of `self.__doc__` is `str | None`, so in the `None` case trying to access the `strip()` method will cause an `AttributeError`. That's not really a problem for this method because it will catch the `AttributeError`, but type checkers don't like using exceptions for control flow like that.

> I don't mind the extra annotation, but I guess it should at least be applied (or left out) consistently.

The majority of `astropy` has been written without taking type checking into account at all. This makes annotating `astropy` is complicated enough even without any additional consistency or coverage requirements.

Reply:
> The majority of astropy has been written without taking type checking into account at all. This makes annotating astropy is complicated enough even without any additional consistency or coverage requirements.

definitely fair enough ! I didn't mean to make your experience even more painful.